srcts/src/components/webcomponents/popover.ts

```diff
@@ -349,43 +349,53 @@ export class BslibPopover extends BslibElement {
   private _updatePopover(data: UpdateMessage): void {
     const { content, header } = data;
 
+    // First, render any HTMLDependency()s
     const deps = [];
     if (content) deps.push(...content.deps);
     if (header) deps.push(...header.deps);
     Shiny.renderDependencies(deps);
 
-    // Since we may need to add a close button (with event handlers),
-    // parse the string into an HTMLElement. And, since .setContent() is less
-    // error-prone if we pass _both_ the header and content, we fallback to the
-    // existing header/content if the user didn't supply one.
-    const getOrCreateElement = (
-      x: typeof content | typeof header,
-      fallback: HTMLElement | undefined,
-      selector: string
-    ): HTMLElement => {
-      if (x) return createWrapperElement(x.html, "contents");
-      if (fallback) return fallback;
-      return this.bsPopover.tip?.querySelector(selector) as HTMLElement;
-    };
+    const { tip } = this.bsPopover;
 
-    const newHeader = getOrCreateElement(
-      header,
-      this.header,
-      ".popover-header"
-    );
-    const newContent = getOrCreateElement(
-      content,
-      this.content,
-      ".popover-body"
-    );
-    render(this._closeButton(newHeader), newContent);
+    // If the user hasn't supplied new content/header, we still need to find and pass
+    // along the current content/header (so they don't get removed on update).
+    // These current elements should always be a <div style="display:contents">
+    // with the actual content inside.
+    const currentHeader = this.visible
+      ? (tip?.querySelector(".popover-header")?.children[0] as HTMLElement)
+      : (this.header as HTMLElement);
```

Comment:
> Do we query tip because changes to the popover content while the popover is open might not be reflected in this.header and this.content?

Yes, but also, when the tip is visible, `this.header` and `this.content` is actually empty because they move from the custom element to the Bootstrap controlled `.popover` container.

Your suggestion is also a little deceiving. We only take the 1st child of the `tip` selection, and we do that to get the value we inserted, which is relevant regardless of whether we wrap the user contents in a `<div style="display:contents">` container div.